tree-sitter-tips.md

```diff
@@ -0,0 +1,79 @@
+# Tree-sitter tips and tricks
+
+Language development using [Tree-Sitter](https://github.com/tree-sitter/tree-sitter) can be an arduous task. These tips are intended to help bring empiricism to the process by reducing trial and error (or at least introducing some guard rails).
+
+### :bulb: Rely on Language Specs, but not too much.
+There are language specifications (like [this one for Java](https://docs.oracle.com/javase/specs/jls/se9/html/index.html)) that comprehensively describe a language. Specs provide the lexical and syntactic structure of a language in addition to its components (ex., types, variables, declarations, etc.) and how they interconnect. This can serve as a great baseline for writing your grammar. However, mirroring the spec 100% is not always the best approach, especially when its structure does not align with the parse tree you'd like to generate using tree-sitter. In this situation, it's better to deviate from the spec and prioritize generating a good parse tree. One way to do this may be to find an LR parser for the language you're working on ([such as this one for Java](https://github.com/javaparser/javaparser)) to see how they've modeled their tree outputs. Studying tree outputs from other LR parsers can be a good way to guide your own grammar development. 
+
+### :bulb: What does a "good" parse tree look like?
+
+Here are some attributes of a "pretty" tree:
```

Comment:
:heart: the idea of clearly defining what a good parse tree looks like. This could be built into sort of a style guide for trees. Similar to an earlier comment, I think defining a "good" parse tree is going to be closely related to what you are going to use the tree for. Maybe we could start with stating some of the goals of tree-sitter and work back to a style guide.